### PR TITLE
SCF bump to v1.15.0, plus subordinate releases

### PR DESCRIPTION
**Pull request by andreas-kupries**
Trello: https://trello.com/c/HHM178x6/572-5-cve-2018-1221-bump-cf-uaa
UFR = https://github.com/SUSE/uaa-fissile-release/pull/80

__Note__: `nfs-volume-release` is not listed in `cf-deployment`. Bumping it deferred at the moment (It could be, there are new releases).

Under vagrant this commit was :white_check_mark: for cluster start up to :cat:, including.

### CODE REVIEW

**Comment by andreas-kupries** [comment]
Not sure where git sees the conflict. The submodule is on https://github.com/SUSE/uaa-fissile-release/pull/80 and this PR is branched from the current master for it ... Hm. Maybe that master is not yet integrated with SCF ?

**Comment by alexjh** [comment]
diego-release v1.34.0 added some new config properties, and I think we should be using locket, since we don't have consul AFAIK.

```
route_emitter and route_emitter_windows

    Added locks.consul.enabled: Whether the route-emitter in global mode uses Consul for its component lock. Defaults to true.
    Added locks.locket.enabled: Whether the route-emitter in global mode uses Locket for its component lock. Defaults to false.
    Added locks.locket.hostname: Hostname of the Locket service. Defaults to locket.service.cf.internal.
    Added locks.locket.enabled: Port of the Locket service. Defaults to 8891.
```

**Comment by alexjh** [comment]
@andreas-kupries I think you'll have to rebase on something past 9c9f4111 because the original UAA submodule pointer changed.

**Comment by andreas-kupries** [comment]
> diego-release v1.34.0 added some new config properties, and I think we should be using locket, since we don't have consul AFAIK.

Note that the description mentions __global mode__. If you look into our opinions you will find that we operate diego in __local__ mode, making the properties moot. I mention that at the bottom of my commit message for 88aa382a11d3af5793ecd9d1c2c74b6bf31383db

File: src/scf-helper-release/packages/cli/packaging [hunk 1]
```diff
@@ -2,7 +2,7 @@ set -e
 
 cd cli
 
-tar xzf cf-cli_6.31.0_linux_x86-64.tgz
+tar xzf cf-cli_6.33.1_linux_x86-64.tgz
```
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply src/scf-helper-release/packages/cli/packaging]: Hm. Sure, that should be possible.

File: bin/get-cf-versions.sh [hunk 2]
```diff
@@ -13,15 +13,23 @@ RELEASE=$1
 # gem install csv2json yaml2json orderedhash
 # brew install jq
 
-CF_RELEASE=https://raw.githubusercontent.com/cloudfoundry/cf-release/master/releases/cf-$RELEASE.yml
-COMMIT_HASH=$(curl $CF_RELEASE 2>/dev/null | yaml2json | jq '"X"+.commit_hash')
+CF_RELEASE=https://raw.githubusercontent.com/cloudfoundry/cf-deployment/v${RELEASE}/cf-deployment.yml
```
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply bin/get-cf-versions.sh]: Done with commit 8e0a18d02b56cdd4c639f197ed1b3625997a08f1

File: bin/update-releases.sh [hunk 3]
```diff
@@ -16,10 +16,34 @@ VERSION_INFO=$("${GIT_ROOT}/bin/get-cf-versions.sh" "${RELEASE}")
 mkdir -p ${GIT_ROOT}/_work
 echo "${VERSION_INFO}" > ${GIT_ROOT}/_work/VERSION_INFO
 
-CF_RELEASE=$(echo "${VERSION_INFO}" | jq -r .[\"cf-release-commit-sha\"])
-DIEGO_RELEASE=v$(echo "${VERSION_INFO}" | jq -r .[\"diego-release-version\"])
-CFLINUXFS2_RELEASE=v$(echo "${VERSION_INFO}" | jq -r .[\"cflinuxfs2-release-version\"])
-GARDEN_RUNC_RELEASE=v$(echo "${VERSION_INFO}" | jq -r .[\"garden-runc-release-version\"])
+release_ref () {
+    release="${1}"
```
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply bin/update-releases.sh]: Done with commit 8e0a18d02b56cdd4c639f197ed1b3625997a08f1

File: bin/update-releases.sh [hunk 4]
```diff
@@ -16,10 +16,34 @@ VERSION_INFO=$("${GIT_ROOT}/bin/get-cf-versions.sh" "${RELEASE}")
...
+    echo "${VERSION_INFO}" | awk -F , "/$release/ { print \$2 }"
+}
+
+release_origin() {
+    release=${1}
```
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply bin/update-releases.sh]: Done with commit 8e0a18d02b56cdd4c639f197ed1b3625997a08f1

File: bin/update-releases.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply bin/update-releases.sh]: Done with commit 8e0a18d02b56cdd4c639f197ed1b3625997a08f1

File: bin/update-releases.sh [hunk 5]
```diff
@@ -71,55 +107,48 @@ do
...
-CAPI_RELEASE=$(get_submodule_ref "${CF_RELEASE_VERSION_INFO}" capi-release)
-CONSUL_RELEASE=$(get_submodule_ref "${CF_RELEASE_VERSION_INFO}" consul-release)
-ETCD_RELEASE=$(get_submodule_ref "${CF_RELEASE_VERSION_INFO}" etcd-release)
-LOGGREGATOR_RELEASE=$(get_submodule_ref "${CF_RELEASE_VERSION_INFO}" loggregator-release)
-NATS_RELEASE=$(get_submodule_ref "${CF_RELEASE_VERSION_INFO}" nats-release)
-
-#CFMYSQL_RELEASE=$(get_submodule_ref "${CF_RELEASE_VERSION_INFO}" cf-mysql-release)
-ROUTING_RELEASE=$(get_submodule_ref "${CF_RELEASE_VERSION_INFO}" routing-release)
-
+# GO_BUILDPACK_RELEASE=$(get_submodule_ref go-buildpack-release)
```
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply bin/update-releases.sh]: Done with commit 8e0a18d02b56cdd4c639f197ed1b3625997a08f1

File: container-host-files/etc/scf/config/role-manifest.yml [hunk 6]
```diff
@@ -644,7 +644,6 @@ roles:
   - scripts/nginx_log_level.sh
   scripts:
   - scripts/patches/fix_blobstore_send_timeout.sh
-  - scripts/patches/fix_duplicate_buildpack_positions.sh
```
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply container-host-files/etc/scf/config/role-manifest.yml]: Nothing to do ;)

File: src/scf-helper-release/packages/cli/packaging (hunk 1 above)
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply src/scf-helper-release/packages/cli/packaging]: Done with commit 8e0a18d02b56cdd4c639f197ed1b3625997a08f1

**Comment by andreas-kupries** [comment]
Rebase done, now based on d6d1ac364750a5b53154ffcfa58144ec0f62eb1b

**Comment by alexjh** [comment]
Resolves 1371.

**Pull request merged**